nandodev-net/api-ooni-data-analytics-python
# feat: Creating a CustomUser app.

## Tickets

## Description
This PR creates the CustomUser functionality, this allows us to create users with the following roles:
- Superuser
- Admin
- Analyst
- Editor
- Guest

## Changelog

- [x] Created the CustomUser model with the class AbstractBaseUser.
- [x] Created the CustomUserManager from BaseUserManager that defines the required fields to create a superuser and a user.
- [x]  Modified the Makefile with better settings.
- [x] Centralize the styling tools settings in one setup.cfg file.
- [x] User app subscribed to the Django Admin Site.


## How To Test

- Install Docker in your system.
- Clone the project and run:

  1. `chmod +x run_docker.sh`
  2. `chmod +x entrypoint.sh`
  3. `./run_docker.sh --dev`
  4. `docker compose -f docker-compose.dev.yml exec api-web python manage.py makemigrations`
  5. `docker compose -f docker-compose.dev.yml exec api-web python manage.py migrate --noinput`
  6. `docker compose -f docker-compose.dev.yml exec api-web python manage.py createsuperuser`
  7. Go to http://localhost:8000/admin/
  8. Login with the created superuser
  9. Verify the existence of the CustomUser model.

![image](https://github.com/nandodev-net/api-ooni-data-analytics-python/assets/12719909/e0c7f816-98bd-4bdd-aa11-8d8f73a9a052)
